test/abstractarray.jl

```diff
@@ -932,7 +932,7 @@ end
 end
 
 @testset "reverse dim on empty" begin
-    @test reverse(Diagonal([]),dims=1) == Diagonal([])
+    @test reverse(Diagonal(Int[]),dims=1) == Diagonal([])
```

Comment:
```suggestion
    @test reverse(Diagonal([]),dims=1) == Diagonal([])
```
Maybe it's better to leave this as-is and use my other suggested fix from my comment — defining a `Matrix(::Diagonal{Any})` method — to prevent the theoretical breaking change here.